velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -224,60 +297,91 @@ HiveDataSink::HiveDataSink(
       insertTableHandle_(std::move(insertTableHandle)),
       connectorQueryCtx_(connectorQueryCtx),
       commitStrategy_(commitStrategy),
+      maxOpenWriters_(
+          HiveConfig::maxPartitionsPerWriters(connectorQueryCtx_->config())),
       partitionChannels_(getPartitionChannels(insertTableHandle_)),
       partitionIdGenerator_(
           !partitionChannels_.empty() ? std::make_unique<PartitionIdGenerator>(
                                             inputType_,
                                             partitionChannels_,
-                                            HiveConfig::maxPartitionsPerWriters(
-                                                connectorQueryCtx_->config()),
+                                            maxOpenWriters_,
                                             connectorQueryCtx_->memoryPool())
-                                      : nullptr) {
+                                      : nullptr),
+      bucketCount_(
+          insertTableHandle_->bucketProperty() == nullptr
+              ? 0
+              : insertTableHandle_->bucketProperty()->bucketCount()),
+      bucketFunction_(
+          isBucketed() ? createBucketFunction(
+                             insertTableHandle_->bucketProperty(),
+                             inputType_)
+                       : nullptr) {
+  VELOX_USER_CHECK(
+      !isBucketed() || isPartitioned(),
+      "A bucket table must be partitioned: {}");
+  if (isBucketed()) {
+    VELOX_USER_CHECK_LT(
+        bucketCount_,
+        maxBucketCount(),
+        "bucketCount {} must be smaller than {}",
```

Comment:
No need to include bucketCount_ and maxBucketCount in the error message. These are included by default.